agent/cache-types/connect_ca_leaf.go

```diff
@@ -108,7 +108,7 @@ func (c *ConnectCALeaf) Fetch(opts cache.FetchOptions, req cache.Request) (cache
 		return result, errors.New("invalid RootCA response type")
 	}
 	if roots.TrustDomain == "" {
-		return result, errors.New("cluster has no CA bootstrapped")
+		return result, errors.New("cluster has no CA bootstrapped yet")
```

Comment:
This used to cover both the case where Connect _was_ enabled but the request raced with initial CA bootstrap and the case where Connect is not enabled at all. Now I've added an explucit error for not-enabled that is more descriptive, the only way (tm) this can be hit is during a race so I think adding "yet" helps to hint that the request was just a bit early and not that anything is misconfigured.